### ISSUE TITLE
Stop training process

### ISSUE THREAD

**Issue by zxmanxz** [CorentinJ/Real-Time-Voice-Cloning#661]
Hi, there! I want to get some kind of automation of synthesizer model training. For example, want to stop training when loss is equal some value. Can you please give me an advise how can I do this?

**Comment by blue-fish** [comment]
The following instructions pertain to the latest repo code, which uses the pytorch synthesizer in #472. We no longer support the tensorflow synth.

Training occurs in a nested loop in [synthesizer/train.py](https://github.com/CorentinJ/Real-Time-Voice-Cloning/blob/master/synthesizer/train.py). You can break out of it when the loss is below your threshold.

```
for epoch in range(1, epochs+1):
    for i, (texts, mels, embeds, idx) in enumerate(data_loader, 1):
        # Loss calculation (line 184)
        loss = m1_loss + m2_loss + stop_loss

        # Loss window (line 197), tracks past 100 loss values
        loss_window.append(loss.item())

        """
        Add these lines to the end of the inside loop
        """
        if loss_window.average < training_stop_threshold:
            model.save(weights_fpath, optimizer)
            break
    """
    Add these lines to the end of the outside loop
    """
    if loss_window.average < training_stop_threshold:
        break
```

**Comment by zxmanxz** [comment]
Thank for your answer.
